tests/contrib/asyncpg/test_asyncpg.py

```diff
@@ -0,0 +1,154 @@
+from typing import Generator
+
+import asyncpg
+import pytest
+import pytest_asyncio
+
+from ddtrace import Pin
+from ddtrace.contrib.asyncpg import patch
+from ddtrace.contrib.asyncpg import unpatch
+from tests.contrib.config import POSTGRES_CONFIG
+
+
+@pytest.fixture(autouse=True)
+def patch_asyncpg():
+    # type: () -> Generator[None, None, None]
+    patch()
+    yield
+    unpatch()
+
+
+@pytest_asyncio.fixture
+async def patched_conn():
+    # type: () -> Generator[asyncpg.Connection, None, None]
+    conn = await asyncpg.connect(
+        host=POSTGRES_CONFIG["host"],
+        port=POSTGRES_CONFIG["port"],
+        user=POSTGRES_CONFIG["user"],
+        database=POSTGRES_CONFIG["dbname"],
+        password=POSTGRES_CONFIG["password"],
+    )
+    yield conn
+    await conn.close()
+
+
+@pytest.mark.asyncio
+async def test_connect(snapshot_context):
+    with snapshot_context():
+        conn = await asyncpg.connect(
+            host=POSTGRES_CONFIG["host"],
+            port=POSTGRES_CONFIG["port"],
+            user=POSTGRES_CONFIG["user"],
+            database=POSTGRES_CONFIG["dbname"],
+            password=POSTGRES_CONFIG["password"],
+        )
+        await conn.close()
+
+    # Using dsn should result in the same trace
+    with snapshot_context():
+        conn = await asyncpg.connect(
+            dsn="postgresql://%s:%s@%s:%s/%s"
+            % (
+                POSTGRES_CONFIG["user"],
+                POSTGRES_CONFIG["password"],
+                POSTGRES_CONFIG["host"],
+                POSTGRES_CONFIG["port"],
+                POSTGRES_CONFIG["dbname"],
+            )
+        )
+        await conn.close()
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot(ignores=["meta.error.stack"])  # stack is noisy between releases
+async def test_bad_connect():
+    with pytest.raises(OSError):
+        conn = await asyncpg.connect(
+            host="localhost",
+            port=POSTGRES_CONFIG["port"] + 1,
+        )
+        await conn.close()
```

Comment:
do this get called if an exception is raised?

do we need to call it if there was an error connecting?

Reply:
No it won't get called if an exception is raised

Reply:
do we need it to be? otherwise if it won't get called, why have it?